asreview/webapp/app.py

```diff
@@ -95,23 +100,20 @@ def create_app(
         def load_user(user_id):
             return User.query.get(int(user_id))
 
+        # Verify email server has been configured
+        app.config["EMAIL_CONFIG"] = all([
+            app.config.get("MAIL_SERVER", False),
+            app.config.get("MAIL_USERNAME", False),
+            app.config.get("MAIL_PASSWORD", False)
+        ])
+
         if app.config.get("EMAIL_VERIFICATION", False) and not app.config.get(
             "EMAIL_CONFIG", False
         ):
             raise ValueError(
                 "Missing email configuration to facilitate email verification"
             )
 
```

Comment:
Shall we remove this completely? If the email is not configured, an error can be raised when sending the email. I think we should avoid checks on the config if they arent covering all the config variables.

Reply:
That was the initial idea, but email-verification as a parameter (being true or false) is sent with /boot to the frontend in which it is used in certain situations (like a forgot-password) link. With the [EMAIL_CONFIG] key lost in the TOML file I can't check anymore if there is a configuration available for emails. This was my poor-man's solution. Ever email config needs a mail server and probably a username and password.

Reply:
I misunderstood maybe. I had doubts about lines 110 and onwards. They can go. But I'd like to keep 103 - 108 since the frontend has to know whether an email service is configured (then it can show the "forgot password" link). Not the most elegant code, but it does the job.

Reply:
Ah, thanks; this was the missing link for me. In that case, I also favor removing lines 110 and onwards. I prefer to remove the config "EMAIL_CONFIG" and replace

```
            response["email_config"] = bool(app.config.get("EMAIL_CONFIG", False))

```
with
```
            response["email_config"] = all([
              app.config.get("MAIL_SERVER", False),
              app.config.get("MAIL_USERNAME", False),
              app.config.get("MAIL_PASSWORD", False)
          ])
```
or create a function `_is_email_configured()` to check for this.

By doing this, nobody can think you can set this config var via env or TOML. This is not something you can config yourself

Reply:
The other 2 references to EMAIL_CONFIG are in auth.py. Same can be applied.

Reply:
I have used your suggestion in app.py. In auth.py I have created a function since I had to check the same thing twice.